core/domain/remote_access_services_test.py

```diff
@@ -58,62 +59,38 @@ def post_callback(request):
         self.assertIn('job_id', resp.keys())
         self.assertIn('algorithm_id', resp.keys())
         self.assertIn('training_data', resp.keys())
+        self.assertIn('algorithm_version', resp.keys())
 
         self.assertEqual(resp['job_id'], '1')
         self.assertEqual(resp['algorithm_id'], 'ab')
+        self.assertEqual(resp['algorithm_version'], 1)
         self.assertDictEqual(resp['training_data'], {})
 
     def test_result_gets_stored_correctly(self):
         """Test that correct results are stored."""
 
-        job_result_dict = {
-            'job_id': '123',
-            'classifier_data_with_floats_stringified': {
-                'param': 'val'
-            }
-        }
+        frozen_model_proto = text_classifier_pb2.TextClassifierFrozenModel()
+        frozen_model_proto.model_json = 'model_json'
+        algorithm_id = 'TextClassifier'
+        job_id = '123'
+        job_result = training_job_result_domain.TrainingJobResult(
+            job_id, algorithm_id, frozen_model_proto)
 
         # Callback for post request.
         @self.callback
         def post_callback(request):
             """Callback for post request."""
-            self.assertEqual(request.payload['message']['job_id'], '123')
-            classifier_data = {
-                'param': 'val'
-            }
-            self.assertDictEqual(
-                classifier_data,
-                request.payload['message'][(
-                    'classifier_data_with_floats_stringified')])
+            payload = (
+                training_job_response_payload_pb2.TrainingJobResponsePayload())
+            payload.ParseFromString(request.body)
+            self.assertEqual(payload.job_result.job_id, '123')
+            self.assertEqual(
+                payload.job_result.WhichOneof('classifier_frozen_model'),
+                'text_classifier')
+            self.assertEqual(
+                payload.job_result.text_classifier.model_json, 'model_json')
 
         with self.set_job_result_post_callback(post_callback):
             status = remote_access_services.store_trained_classifier_model(
-                job_result_dict)
+                job_result)
         self.assertEqual(status, 200)
-
-    def test_exception_is_raised_when_classifier_data_is_inappropriate(self):
```

Comment:
Just to check, why was this deleted? Isn't it still possible for job_result.validate() to fail somehow?

(I know you have tests for it specifically, but the point is to make sure that validate() is called from this function too in order to ensure that invalid models can't be saved.)

Reply:
Added a complimentary test case for this, it does not do full set of validation checks but just ensures that validate has been called.